client/client.go

```diff
@@ -462,6 +462,46 @@ func (r *NotaryRepository) GetTargetByName(name string, roles ...string) (*Targe
 
 }
 
+// CheckTargetByName returns true if all passed-in roles have signed the specified target with the same hash.
+// If no roles are passed, it uses the targets role.  Note that child roles are considered if
+// we can't find the target in the specified role.
+func (r *NotaryRepository) CheckTargetByName(name string, roles ...string) (bool, error) {
+	if err := r.Update(false); err != nil {
+		return false, err
+	}
+
+	if len(roles) == 0 {
```

Comment:
Totally non-blocking - wondering it's worth filtering out roles that are parents of other roles.  E.g. if you specify `targets` and also `targets/releases`, there is no need to search `targets` because if it exists in `targets/releases` we count that as also being in `targets`.

Unless we want to error because maybe the user means that someone signed it specifically into `targets/releases` and also signed it into `targets` itself?